packages/client/src/connection.ts

```diff
@@ -291,11 +304,23 @@ export class Connection {
     optionsWithDefaults.metadata['client-name'] ??= 'temporal-typescript';
     optionsWithDefaults.metadata['client-version'] ??= pkg.version;
 
+    const originalGrpcProxy = process.env.grpc_proxy;
+    if (normalizedOptions.proxy) {
+      const { targetHost: target, basicAuth: auth } = normalizedOptions.proxy;
+      const { scheme, hostname: host, port } = parseHttpConnectProxyAddress(target);
+      const authString = auth ? `${auth.username}:${auth.password}@` : '';
+      process.env.grpc_proxy = `${scheme}://${authString}${host}:${port}`;
```

Comment:
Hrmm, I still think we should leave the env alone and just tell people they should use it if they want proxying. Note, this is what happens in Go. Granted we have no synchronous guarantees that our code doesn't get interrupted so we couldn't do this there if we wanted. But still, users are used to the idea of needing an environment variable for this and if our gRPC library chooses to make it more explicit later, we can do the same.